Linear_cell_complex/demo/Linear_cell_complex/Viewer.h

```diff
@@ -42,20 +51,34 @@ class Viewer : public QGLViewer
   GLuint m_dlEdges;
   GLuint m_dlVertices;
   bool m_displayListCreated;
+  bool m_previous_scene_empty;
+  
+  GLUtesselator* FTess;
 
   typedef LCC::Dart_handle Dart_handle;
   typedef LCC::Dart_const_handle Dart_const_handle;
 
-
 public:
   Viewer(QWidget* parent)
     : QGLViewer(parent), wireframe(false), flatShading(true),
-      edges(true), vertices(true), m_displayListCreated(false)
+      edges(true), vertices(true), m_displayListCreated(false),
+      m_previous_scene_empty(true)
   {
     QGLFormat newFormat = this->format();
     newFormat.setSampleBuffers(true);
     newFormat.setSamples(16);
     this->setFormat(newFormat);
+
+    FTess = gluNewTess();
+    gluTessCallback(FTess, GLU_TESS_BEGIN,(void (*) ( ))  & glBegin);
```

Comment:
I agree, these code need to be modified when moving to new OpenGL code.
Note that in some application, I really need to draw non convex faces.